src/Symfony/Bundle/FrameworkBundle/Resources/config/services.xml

```diff
@@ -12,12 +12,18 @@
         </service>
         <service id="Symfony\Component\EventDispatcher\EventDispatcherInterface" alias="event_dispatcher" />
 
-        <service id="http_kernel" class="Symfony\Component\HttpKernel\HttpKernel" public="true">
+        <service id="Symfony\Component\HttpKernel\HttpKernel">
             <argument type="service" id="event_dispatcher" />
             <argument type="service" id="controller_resolver" />
             <argument type="service" id="request_stack" />
             <argument type="service" id="argument_resolver" />
         </service>
+        <service id="Symfony\Component\HttpKernel\Middleware\ServiceResetMiddleware">
+            <argument key="$httpKernel" type="service" id="Symfony\Component\HttpKernel\HttpKernel" />
+            <argument key="$services" /> <!-- ResettableServicePass will inject an iterator of initialized services here ($serviceId => $serviceInstance) -->
+            <argument key="$resetMethods" type="collection" /> <!-- ResettableServicePass will inject an array of reset methods here ($serviceId => $method) -->
+        </service>
+        <service id="http_kernel" alias="Symfony\Component\HttpKernel\Middleware\ServiceResetMiddleware" public="true" />
```

Comment:
Rather than doing this (which makes the HttpKernel typehint autowirable btw), I would rather keep the http_kernel as it is currently, and then use service decoration when the reset middleware is needed

Reply:
Since we do not implement a switch for turning off that feature (which was the whole point of moving the reset to the beginning), the middleware would technically be always present.

Reply:
But I agree that enabling `HttpKernel` for autowiring could cause accidents. ;-)